crates/weaver_live_check/src/lib.rs

```diff
@@ -0,0 +1,341 @@
+use sample_span::{SampleSpan, SampleSpanEvent, SampleSpanLink};
+use serde::{Deserialize, Serialize};
+use weaver_checker::violation::{Advice, AdviceLevel};
+use weaver_common::{
+    diagnostic::{DiagnosticMessage, DiagnosticMessages},
+    Logger,
+};
+use weaver_forge::registry::ResolvedRegistry;
+
+/// Advisors for live checks
+pub mod advice;
+/// An ingester that reads samples from a JSON file.
+pub mod json_file_ingester;
+/// An ingester that reads samples from standard input.
+pub mod json_stdin_ingester;
+/// Live checker
+pub mod live_checker;
+/// The intermediary format for attributes
+pub mod sample_attribute;
+/// An intermediary format for resources
+pub mod sample_resource;
+/// The intermediary format for spans
+pub mod sample_span;
+/// An ingester that reads attribute names from a text file.
+pub mod text_file_ingester;
+/// An ingester that reads attribute names from standard input.
+pub mod text_stdin_ingester;
+
+/// Missing Attribute advice type
+pub const MISSING_ATTRIBUTE_ADVICE_TYPE: &str = "missing_attribute";
+/// Template Attribute advice type
+pub const TEMPLATE_ATTRIBUTE_ADVICE_TYPE: &str = "template_attribute";
+
+/// Weaver live check errors
+#[derive(thiserror::Error, Debug, Clone, PartialEq, Serialize, Diagnostic)]
+#[non_exhaustive]
+pub enum Error {
+    /// Generic ingest error.
+    #[error("Fatal error during ingest. {error}")]
+    IngestError {
+        /// The error that occurred.
+        error: String,
+    },
+
+    /// Attempt to Ingest an empty line.
+    #[error("Attempt to ingest an empty line.")]
+    IngestEmptyLine,
+
+    /// Advice error.
+    #[error("Fatal error from Advisor. {error}")]
+    AdviceError {
+        /// The error that occurred.
+        error: String,
+    },
+
+    /// Output error.
+    #[error("Output error. {error}")]
+    OutputError {
+        /// The error that occurred.
+        error: String,
+    },
+}
+
+impl From<Error> for DiagnosticMessages {
+    fn from(error: Error) -> Self {
+        DiagnosticMessages::new(vec![DiagnosticMessage::new(error)])
+    }
+}
+
+/// Ingesters implement a trait that returns an iterator of samples
+pub trait Ingester {
+    /// Ingest data and return an iterator of the output type
+    fn ingest(
+        &self,
+        logger: impl Logger + Sync + Clone,
+    ) -> Result<Box<dyn Iterator<Item = Sample>>, Error>;
+}
+
+/// Represents a sample entity
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum Sample {
+    /// A sample attribute
+    Attribute(SampleAttribute),
+    /// A sample span
+    Span(SampleSpan),
+    /// A sample span event
+    SpanEvent(SampleSpanEvent),
+    /// A sample span link
+    SpanLink(SampleSpanLink),
+    /// A sample resource
+    Resource(SampleResource),
+}
+
+/// Represents a sample entity with a reference to the inner type
+#[derive(Debug, Clone, PartialEq, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum SampleRef<'a> {
```

Comment:
Given how small this is - You can likely pass-by-value to Advice. I'll add comments elsewhere, where I think it can work.